src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -11,6 +11,8 @@ public AsyncRequestProcessor(Executor executor) {
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        UserData userData = new UserData(
+                userId, "Details for " + userId);
+        return CompletableFuture.supplyAsync(() -> userData, executor);
     }
```

Comment:
The method `processRequest` is currently creating a new `UserData` object synchronously before supplying it to the `CompletableFuture`. If the creation of `UserData` is a heavy or time-consuming operation, this could block the calling thread. Consider moving the creation of `UserData` inside the lambda passed to `supplyAsync` to ensure it's executed asynchronously.